### DISCUSSION TITLE
API authorization bearer token

### DISCUSSION THREAD

**Discussion by scastellanosnqn** [polymind-inc/acmebot#529] [Q&A] [answered]
Hello, I'm wondering if it is possible to access the API externally via postman (for example) in order to query existing certificates or create new ones. 
Maybe there is some documentation on how the API works or at least how to authenticate via postman/curl ?
I'm trying to make some automation based on the already working Function App, I've tried to set my bearer token (taken from: https://resources.azure.com/api/token?plaintext=true) as authorization header without success.

Can anyone point me on the right direction ?

Regards.

**Comment by shibayan** [comment, marked as answer]
Acmebot has Azure AD authentication enabled, so it requires a completely different Bearer Token than the one obtainable in Azure Resource Explorer.

The following documentation should be helpful.
https://learn.microsoft.com/en-us/azure/active-directory/develop/scenario-daemon-overview

**Reply by scastellanosnqn** [reply]
Hi shibayan, thanks for answering, I've quite limited access to our Azure infrastructure, but I currently have access to the Function App and its dashboard.
Is possible to obtain the correct token with no admin access to Azure ? Maybe some az cli command ?

**Reply by shibayan** [reply]
Some level of permissions to Azure AD should be required. Creation of service principals and operating privileges will be required.
